backoffcache.go

```diff
@@ -139,7 +140,11 @@ func (d *BackoffDiscovery) FindPeers(ctx context.Context, ns string, opts ...dis
 		}
 		pch := make(chan peer.AddrInfo, chLen)
 		for _, ai := range c.prevPeers {
-			pch <- ai
+			select {
+			case pch <- ai:
+			default:
+				// skip if we have asked for a lower limit than the number of peers known
```

Comment:
I'm wondering if it's ok here to return more than `options.Limit` values. The consumer could rely on this limit, as it's documented here:
https://github.com/libp2p/go-libp2p-core/blob/525a0b13017263bde889a3295fa2e4212d7af8c5/discovery/options.go#L35-L41

Reply:
Can you clarify your comment, it seems ambiguous and I am not sure if you are asking if it's ok to do something different to this code or if this code is ok as is. This code respects the limit by skipping any that exceed the number requested.

Reply:
Sorry, I don't understand any more what I was saying 4 weeks ago either ;)